quickwit-config/src/source_config.rs

```diff
@@ -255,6 +265,37 @@ impl TryFrom<KinesisSourceParamsInner> for KinesisSourceParams {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct LogRotateSourceParams {
+    /// Path of the current log file.
+    #[serde(deserialize_with = "absolute_filepath_from_str")]
+    pub current_file: PathBuf,
+    /// Path of the directory where rotated logs are stored. If None, assumes it's
+    /// the directory where `latest_file` is.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    #[serde(deserialize_with = "absolute_filepath_from_str_opt")]
+    pub archive_dir: Option<PathBuf>,
+    /// glob pattern of logs that already got rotated
+    pub name_pattern: String,
+}
+
+impl LogRotateSourceParams {
+    pub fn archive_dir(&self) -> PathBuf {
+        if let Some(dir) = &self.archive_dir {
+            dir.clone()
+        } else {
+            let mut dir = self.current_file.clone();
```

Comment:
[file_name](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.file_name)
does that I think?

Reply:
actually that seems to be the opposite, but [`parent`](https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html#method.parent) is what I should have used